### PR TITLE
Add the implementation of API level policies feature

### PR DESCRIPTION
**Pull request by lakshithagunasekara**
This PR adds the implementation of API level policies feature. 

Public git issue: https://github.com/wso2/api-manager/issues/1368

### Related PRs: 
UI - https://github.com/wso2/apim-apps/pull/459
Policy template: https://github.com/wso2/product-apim/pull/13209

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java [hunk 1]
```diff
@@ -444,6 +448,24 @@ public static API importApi(String extractedFolderPath, APIDTO importedApiDTO, B
...
+            apiPoliciesList = OperationPolicyMappingUtil
+                    .fromDTOToAPIOperationPoliciesList(importedApiDTO.getApiPolicies());
+            Map<String, OperationPolicySpecification> visitedPoliciesMap = new HashMap<>();
+            for (OperationPolicy policy : apiPoliciesList) {
+                validateAppliedPolicy(policy, visitedPoliciesMap, extractedFolderPath, apiUUID, provider,
+                        tenantDomain, apiType);
+            }
+
+        }
+        importedApiDTO.setApiPolicies(null);
```
Comment (msm1992) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java:465]: Is this a typo ? What is the need to set apiPolicies of apiDTO to null ?

Comment (lakshithagunasekara) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java:465]: We extract, validate and drop the policies from the API object as to record policy mapping, we need api uuid. API will be created without policies and after that api will be updated.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -1472,123 +1470,149 @@ private void validateAndSetTransports(APIProduct apiProduct) throws APIManagemen
...
-                                            "A common policy is found for " + policyId + ". Validating the policy");
-                                }
+    /**
+     * This method will validate a list of policies and throw error if validation fails
+     * @param apiPoliciesList   Policy list
+     * @param api               API object
+     * @param tenantDomain      Tenant domain
+     * @throws APIManagementException
+     */
+    public List<OperationPolicy> validatePolicies(List<OperationPolicy> apiPoliciesList, API api,
```
Comment (RakhithaRR) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java:1515]: Shall we make this private if it has internal usages only?

**Comment by ashera96** [comment]
Closing this PR since the same is added via [1] alongside minor bug fixes.

1. https://github.com/wso2/carbon-apimgt/pull/12061

**Pull request closed without merging**